kroxylicious-schema-validation/src/main/java/io/kroxylicious/proxy/filter/schema/config/TopicMatchingRecordValidationRule.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.filter.schema.config;
+
+import java.util.Objects;
+import java.util.Set;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/**
+ * Configuration that links a topic name pattern with some record validation rules.
+ */
+public class TopicMatchingRecordValidationRule extends RecordValidationRule {
+
+    private final Set<String> topicNames;
+
+    /**
+     * Construct a new TopicPatternRecordValidationRule
+     * @param topicNames required topic name regex pattern, topics with names matching this pattern are eligible to have this rule applied to them
```

Comment:
You talk about regex here, but later I see topicNames::contains as the matching predicate.  I do think allowing topic names to be specified using regex's is a good idea, as that will match the familiar capabilities of the kafka consumer and allow support for hierarchical topic names.

Reply:
Have removed the reference to regex for now.

Reply:
That's good enough for now.